ee/session_recordings/playlist_counters/recordings_that_match_playlist_filters.py

```diff
@@ -458,15 +490,23 @@ def count_recordings_that_match_playlist_filters(playlist_id: int) -> None:
 
 
 def enqueue_recordings_that_match_playlist_filters() -> None:
-    all_playlists = (
-        SessionRecordingPlaylist.objects.filter(
-            deleted=False,
-            filters__isnull=False,
-        )
-        .filter(Q(last_counted_at__isnull=True) | Q(last_counted_at__lt=timezone.now() - timedelta(hours=2)))
-        .order_by(F("last_counted_at").asc(nulls_first=True))[: settings.PLAYLIST_COUNTER_PROCESSING_PLAYLISTS_LIMIT]
-    )
+    base_query = SessionRecordingPlaylist.objects.filter(
+        deleted=False,
+        filters__isnull=False,
+    ).filter(Q(last_counted_at__isnull=True) | Q(last_counted_at__lt=timezone.now() - timedelta(hours=2)))
+
+    total_playlists_count = base_query.count()
+
+    all_playlists: list[int] = base_query.order_by(F("last_counted_at").asc(nulls_first=True)).values_list(
```

Comment:
doing this made me realise we were loading the entire model from the DB, parsing it, and then just reading the ID

so we can load instead only the ID

and then combined with the slice means we load a small list of ints into memory instead of a big list of django model objects

Reply:
so now we'll run two queries based on the base query

1. select count(*) from blah
2. select id from blah

meaning it's pretty cheap to count the total and then trigger all the tasks for this "page" of data

Reply:
added postgres snapshots to check my memory of how django ORM works